saleor/graphql/product/schema.py

```diff
@@ -251,8 +252,12 @@ def resolve_attribute(self, info, id):
     def resolve_categories(self, info, level=None, **kwargs):
         return resolve_categories(info, level=level, **kwargs)
 
-    def resolve_category(self, info, id):
-        return graphene.Node.get_node_from_global_id(info, id, Category)
+    def resolve_category(self, info, id=None, slug=None):
```

Comment:
I think we should implement this function based on `resolve_menu`.

Reply:
In `resolve_page` and `resolve_menu`, we prioritize queries by slug. Why we change it to prioritize by id?

Reply:
We should agree to one behavior of this type of function because now if you pass invalid `slug` to `resolve_page` system try to resolve object by id. In `resolve_menu` if we don't find an object by `name` system don't try to resolve object by id.
@maarcingebala Which solution is better?

Reply:
Why do we prioritize `name` instead of `menu_id`? `Name` is mutable, `id` is not.

Reply:
Unified and added validation for inputs